extensions/panache/hibernate-orm-rest-data-panache/deployment/src/test/java/io/quarkus/hibernate/orm/rest/data/panache/deployment/AbstractGetMethodTest.java

```diff
@@ -127,4 +128,96 @@ void shouldListComplexHalObjects() {
                 .and().body("_links.add.href", endsWith("/collections"))
                 .and().body("_links.list.href", endsWith("/collections"));
     }
+
+    @Test
+    void shouldNotGetNonExistentPage() {
+        given().accept("application/json")
+                .and().queryParam("page", 100)
+                .when().get("/items")
+                .then().statusCode(200)
+                .and().body("id", is(empty()));
+    }
+
+    @Test
+    void shouldNotGetNegativePageOrSize() {
+        given().accept("application/json")
+                .and().queryParam("page", -1)
+                .and().queryParam("size", -1)
+                .when().get("/items")
+                .then().statusCode(200)
+                // Invalid page and size parameters are replaced with defaults
+                .and().body("id", contains(1, 2));
+    }
+
+    @Test
+    void shouldGetFirstPage() {
+        given().accept("application/json")
+                .and().queryParam("page", 0)
+                .and().queryParam("size", 1)
+                .when().get("/items")
+                .then().statusCode(200)
+                .and().body("id", contains(1))
+                .and().body("name", contains("first"))
+                .and().header("first", endsWith("/items?page=0&size=1"))
```

Comment:
Are you sure those are good headers? I though user headers had to start with `X-`?

Reply:
Actually TIL `X-` prefixes are deprecated now: https://tools.ietf.org/html/rfc6648

Still, shouldn't this rather use the `Link` header? https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link

Reply:
I wasn't sure about the header name and was hoping that somebody would give a suggestion in the PR. `Link` header makes sense of course. What about `first`, `last`, `prev`, `next` as rel?

Reply:
That's what https://tools.ietf.org/html/rfc5005#section-3 used, so there's precedence. Unless you find something standardish more modern?